src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.nextElement = newNode;
+        }
+        tail = newNode;
+
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > 0 && index < size) {
+            Node<T> atIndexNode = head;
+            for (int i = 0; i < index; i++) {
+                atIndexNode = atIndexNode.nextElement;
+            }
+            atIndexNode.prevElement.nextElement
```

Comment:
atIndexNode.prev  should reference you new node, but it still references old node